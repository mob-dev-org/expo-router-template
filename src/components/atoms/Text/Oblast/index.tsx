import { StyleSheet, ViewStyle } from "react-native";
import { View } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import colors, { getColorWithOpacity } from "@src/constants/colors";
import Text from "..";
import Color from "color";

type OblastComponentProps = {
  oblast: any;
  style?: ViewStyle;
  focused: boolean;
};
export default function OblastComponent(props: OblastComponentProps) {
  const { oblast, style, focused } = props;
  return (
    <View
      style={[
        styles.category,
        {
          backgroundColor: colors.primary,
        },
        style,
        !focused
          ? {
              backgroundColor: colors.background,
              borderWidth: 1,
              borderColor: colors.lightBlue,
            }
          : {
              borderWidth: 1,
              borderColor: colors.darkBlue,
            },
      ]}>
      <Text
        size={14}
        style={{ letterSpacing: 0.5 }}
        color={focused ? "white" : undefined}
        weight={focused ? "600" : "400"}>
        {oblast?.name}
      </Text>
    </View>
  );
}
const styles = StyleSheet.create({
  category: {
    padding: 8,
    paddingHorizontal: 16,
    borderRadius: 6,
    alignItems: "center",
    flexDirection: "row",
    justifyContent: "center",
  },
});

// <Ionicons name="list" size={16} color={focused ? "white" : "black"} />;
