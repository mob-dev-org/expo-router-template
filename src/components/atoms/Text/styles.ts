import { StyleSheet } from 'react-native';

// import { color } from '../../../const/variables';

export default StyleSheet.create({
    black: {
        // color: color.black,
    },
    text: {
        // color: color.fontPrimary,
    },
    white: {
        // color: color.fontSecondary,
    },
    blueGray: {
        color: '#A4B2CE',
    },
    gray: {
        color: '#445866',
    },
    gray2: {
        color: '#767A83',
    },
    lightGray: {
        color: '#A4B2CE',
    },
    green: {
        // color: color.newGreen,
    },
    red: {
        // color: color.newRed,
    },
    blue: {
        // color: color.blue,
    },
    lightBlue: {
        // color: color.lightBlue,
    },
    lightGrayWithOpacity: {
        color: '#071C2C66',
    },
    light: {
        fontFamily: 'Poppins-Light',
    },
    regular: {
        fontFamily: 'Poppins-Regular',
    },
    medium: {
        fontFamily: 'Poppins-Medium',
    },
    semiBold: {
        fontFamily: 'Poppins-SemiBold',
    },
    bold: {
        fontFamily: 'Poppins-Bold',
    },
    uppercase: {
        textTransform: 'uppercase',
    },
});
