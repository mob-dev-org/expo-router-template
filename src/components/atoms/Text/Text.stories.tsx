import Text, { TextProps } from './index';

export default {
    title: 'Text',
    component: Text,
    argTypes: {
        weight: {
            options: ['regular', 'medium', 'semiBold', 'bold'],
            control: { type: 'select' },
        },
        center: {
            control: { type: 'boolean' },
        },
        uppercase: {
            control: { type: 'boolean' },
        },
        color: {
            options: [
                'black',
                'white',
                'blueGray',
                'gray',
                'gray2',
                'lightGray',
                'green',
                'red',
                'blue',
                'lightBlue',
                'lightGrayWithOpacity',
            ],
            control: { type: 'select' },
        },
        themeColor: {
            options: [
                'papir',
                'plastikametal',
                'staklo',
                'mjesoviti',
                'bio',
                'glomazni',
                'otok',
                'dvoriste',
                'tekstil',
                'ostalo',
            ],
            control: { type: 'select' },
        },
    },
};

export const Typography = (args: TextProps) => {
    return (
        <>
            <Text {...args} size={32}>
                Typography size:32
            </Text>

            <Text {...args} size={24}>
                Typography size:24
            </Text>

            <Text {...args} size={20}>
                Typography size:20
            </Text>

            <Text {...args} size={18}>
                Typography size:18
            </Text>

            <Text {...args} size={16}>
                Typography size:16
            </Text>

            <Text {...args} size={14}>
                Typography size:14
            </Text>

            <Text {...args} size={13}>
                Typography size:13
            </Text>

            <Text {...args} size={12}>
                Typography size:12
            </Text>

            <Text {...args} size={11}>
                Typography size:11
            </Text>

            <Text {...args} size={8}>
                Typography size:8
            </Text>

            <Text {...args} weight="bold" size={20}>
                Typography Bold : Figma 700
            </Text>

            <Text {...args} weight="semiBold" size={20}>
                Typography SemiBold : Figma 600
            </Text>

            <Text {...args} weight="medium" size={20}>
                Typography Medium : Figma 500
            </Text>

            <Text {...args} weight="regular" size={20}>
                Typography Regular : Figma 400
            </Text>
        </>
    );
};

Typography.args = {
    weight: 'regular',
    color: 'black',
    uppercase: false,
};
