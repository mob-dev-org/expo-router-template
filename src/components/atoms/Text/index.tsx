import React, { FunctionComponent } from 'react';
import { Text as RNText, TextStyle } from 'react-native';

// import { themeColor as themeColors } from '../../../const/variables';
// import { GarbageType } from '../../../api/zgApi.schemas';

import styles from './styles';

export type TextColors =
    | 'black'
    | 'white'
    | 'blueGray'
    | 'gray'
    | 'gray2'
    | 'lightGray'
    | 'green'
    | 'red'
    | 'blue'
    | 'lightBlue'
    | 'lightGrayWithOpacity';

export type TextSize = 8 | 11 | 12 | 13 | 14 | 16 | 18 | 20 | 24 | 32;

export type TextProps = {
    size?: TextSize;
    color?: TextColors;
    customColor?: string;
    // TODO: add more weights with numbers in name
    weight?: 'light' | 'regular' | 'medium' | 'semiBold' | 'bold';
    uppercase?: boolean;
    // themeColor?: GarbageType;
    center?: boolean;
    lineHeight?: number;
    letterSpacing?: number;
    numberOfLines?: number;
    style?: TextStyle;
    disableFontScaling?: boolean;
    children: React.ReactNode;
};

const Text: FunctionComponent<TextProps> = ({
    children,
    color = 'black',
    customColor,
    size = 16,
    weight = 'regular',
    uppercase,
    // themeColor,
    center,
    lineHeight,
    numberOfLines,
    style,
    letterSpacing,
    disableFontScaling,
}) => {
    return (
        <RNText
            numberOfLines={numberOfLines}
            style={[
                {
                    ...style,
                    fontSize: size,
                    textAlignVertical: 'center',
                    includeFontPadding: false,
                },
                styles[weight],
                styles[color],
                !!customColor && { color: customColor },
                // themeColor && { color: themeColors[themeColor] },
                uppercase && styles.uppercase,
                center && { textAlign: 'center' },
                !!lineHeight && { lineHeight },
                !!letterSpacing && { letterSpacing },
            ]}
            allowFontScaling={!disableFontScaling}>
            {children}
        </RNText>
    );
};

export default Text;
