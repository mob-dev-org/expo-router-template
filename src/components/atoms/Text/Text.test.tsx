import React from 'react';
import { render } from '@testing-library/react-native';
import Text from './index';

describe('Text', () => {
    it.todo('Create tests other than snapshot ones');

    it('No text', () => {
        const wrapper = render(<Text></Text>);
        expect(wrapper).toMatchSnapshot();
    });

    it('With text', () => {
        const wrapper = render(<Text>With text</Text>);
        expect(wrapper).toMatchSnapshot();
    });
    it('With text and size', () => {
        const wrapper = render(<Text size={10}>With text</Text>);
        expect(wrapper).toMatchSnapshot();
    });
    it('With text and weight bold', () => {
        const wrapper = render(<Text weight="bold">With text</Text>);
        expect(wrapper).toMatchSnapshot();
    });

    it('With text and centered text', () => {
        const wrapper = render(<Text center>With text</Text>);
        expect(wrapper).toMatchSnapshot();
    });

    it('With text and vertically centered text', () => {
        const wrapper = render(<Text centerVertical>With text</Text>);
        expect(wrapper).toMatchSnapshot();
    });

    it('With all of the properties passed', () => {
        const wrapper = render(
            <Text
                color={'red'}
                key={'key'}
                numberOfLines={1}
                size={10}
                style={{ color: 'green' }}
                weight={'bold'}
                center
                centerVertical>
                With text
            </Text>,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
