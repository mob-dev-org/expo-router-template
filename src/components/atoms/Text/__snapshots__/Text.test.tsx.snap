// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text No text 1`] = `
<Text
  style={
    [
      {
        "color": "text",
        "fontSize": 16,
        "fontWeight": "normal",
        "includeFontPadding": false,
      },
      {},
      false,
      false,
    ]
  }
/>
`;

exports[`Text With all of the properties passed 1`] = `
<Text
  numberOfLines={1}
  style={
    [
      {
        "color": "red",
        "fontSize": 10,
        "fontWeight": "bold",
        "includeFontPadding": false,
      },
      {
        "color": "green",
      },
      {
        "textAlign": "center",
      },
      {
        "textAlignVertical": "center",
      },
    ]
  }
>
  With text
</Text>
`;

exports[`Text With text 1`] = `
<Text
  style={
    [
      {
        "color": "text",
        "fontSize": 16,
        "fontWeight": "normal",
        "includeFontPadding": false,
      },
      {},
      false,
      false,
    ]
  }
>
  With text
</Text>
`;

exports[`Text With text and centered text 1`] = `
<Text
  style={
    [
      {
        "color": "text",
        "fontSize": 16,
        "fontWeight": "normal",
        "includeFontPadding": false,
      },
      {},
      {
        "textAlign": "center",
      },
      false,
    ]
  }
>
  With text
</Text>
`;

exports[`Text With text and size 1`] = `
<Text
  style={
    [
      {
        "color": "text",
        "fontSize": 10,
        "fontWeight": "normal",
        "includeFontPadding": false,
      },
      {},
      false,
      false,
    ]
  }
>
  With text
</Text>
`;

exports[`Text With text and vertically centered text 1`] = `
<Text
  style={
    [
      {
        "color": "text",
        "fontSize": 16,
        "fontWeight": "normal",
        "includeFontPadding": false,
      },
      {},
      false,
      {
        "textAlignVertical": "center",
      },
    ]
  }
>
  With text
</Text>
`;

exports[`Text With text and weight bold 1`] = `
<Text
  style={
    [
      {
        "color": "text",
        "fontSize": 16,
        "fontWeight": "bold",
        "includeFontPadding": false,
      },
      {},
      false,
      false,
    ]
  }
>
  With text
</Text>
`;
